extensions/amp-video-docking/0.1/amp-video-docking.js

```diff
@@ -1234,10 +1191,27 @@ export class VideoDocking {
   /**
    * @param {!AmpElement|!VideoOrBaseElementDef} element
    * @param {number=} minRatio
+   * @param {!LayoutRectDef=} opt_clientRect
    * @return {boolean}
    */
-  isVisible_(element, minRatio = 1) {
-    const intersectionRatio = this.calculateIntersectionRatio_(element);
+  isVisible_(element, minRatio = 1, opt_clientRect) {
+    const inlineRect = opt_clientRect || this.getScrollAdjustedRect_(element);
+
+    // If the component's box is way out of view, the runtime might have
+    // discarded its size values.
+    if (!isSizedRect(inlineRect)) {
```

Comment:
FYI, you can now use `ResizeObserver`. Dima created a really nice helper

https://github.com/ampproject/amphtml/blob/8887dffd682439d7ab0bd2aa4e7408d3d126b474/src/utils/size-observer.js#L32-L36

Reply:
Thanks, I'll give that a try later. We might not need this check after all now since all rects come directly from browser.